packages/core/src/PlayerContextProvider.js

```diff
@@ -591,10 +601,41 @@ export class PlayerContextProvider extends Component {
     this.media.load();
   }
 
+  reloadActiveTrack({ shouldPlay = false } = {}) {
+    const { playlist } = this.props;
+    const { activeTrackIndex, paused, currentTime } = this.state;
+    if (!isPlaylistValid(playlist)) {
+      return;
+    }
+    this.goToTrack({
+      track: playlist[activeTrackIndex],
+      shouldForceLoad: true,
+      shouldPlay: shouldPlay || !paused,
+      startingTime: currentTime,
+      index: activeTrackIndex
+    });
+  }
+
+  handlePlayerOnlineAfterFailure() {
+    this.reloadActiveTrack();
+    window.removeEventListener('online', this.handlePlayerOnlineAfterFailure);
+  }
+
   handleTrackPlaybackFailure(event) {
+    const shouldPlay = !this.state.paused;
     this.setState({
       mediaCannotPlay: true
     });
+    const error = event.target.error;
+    // If there's a network error, the current track should be reloaded
+    // in order to recover from the error state the player is in.
+    if (event.target === this.media && error && error.code === 2) {
```

Comment:
Can we have a constant for the error code somewhere so we know which error type `2` is referencing?